packages/itwinui-css/src/surface/surface.scss

```diff
@@ -34,12 +34,13 @@ $surface-padding-with-gutter: var(--iui-size-2xs);
   overflow-y: auto;
 
   &[data-iui-padded='true'] {
-    scrollbar-gutter: stable both-edges;
-    padding-inline: $surface-padding-with-gutter;
-    padding-block: $surface-padding;
-
-    @supports not (scrollbar-gutter: stable) {
-      padding-inline: $surface-padding;
-    }
+    padding: $surface-padding;
+
+    // Scrollbars might not always occupy space even with a stable gutter, so this can still
+    // result in (tolerable) differences on the right side, depending on scrollbar visibility.
+    // More importantly, the left side will remain aligned, because we are not using a gutter there.
+    // See: https://github.com/iTwin/iTwinUI/issues/1801
+    scrollbar-gutter: stable;
+    padding-inline-end: $surface-padding-with-gutter;
```

Comment:
@FlyersPh9 Can you verify that this is what we want? I used a smaller padding on the right side, but we could even keep the larger one.

The smaller padding can potentially be too small for overlay scrollbars, while the larger padding can potentially be too large for non-overlay scrollbars (more common on Chrome Windows, which has the largest share of our users).

https://github.com/iTwin/iTwinUI/assets/9084735/13bf0151-0634-4071-9f20-3ae043f4ec3c

Reply:
Should we have some JS solution to detect whether space is given to the scrollbar or not?

e.g. when space is given to a scrollbar + `scrollbar-gutter: stable`, I think `clientWidth < offsetWidth`. And when no space is given, `clientWidth === offsetWidth`.

Reply:
JS is completely unnecessary here. Neither of the two issues I brought up are bugs or dealbreakers; I'm just asking for opinions which one we want to optimize for.

Reply:
I mean to say that we can optimize both with a JS solution.

i.e. use smaller `inline-end` padding when there the scrollbar takes up space and use larger/regular padding when there the scrollbar takes up no space. That way, the end padding always stays consistent regardless of whether the scrollbar is given space.

Reply:
Can we please not overcomplicate it with unnecessary JS? We can't always optimize for every situation, and I've mentioned multiple times that this inconsistency is tolerable. Simplicity is a vastly more important goal than consistency.

Reply:
Yes, I know this JS is extra weight and overcomplicates the small issue. My comment was just a suggestion, not a requirement.

Out of the two padding options, I prefer the option you used currently. Having smaller than usual inline-end padding when the scrollbar has no space (overlay) is not too much of a deal breaker, since like you mentioned, most of our users have scrollbars that have space.

Reply:
I do prefer to keep this CSS only at least for this fix now.

Screenshots of https://itwin.github.io/iTwinUI/1810/css/surface.html

**Chrome Canary 123:**
![Screenshot 2024-01-31 at 3 00 09 PM](https://github.com/iTwin/iTwinUI/assets/849817/13bc09e0-e1bd-4699-b1e5-9732aac2609b)

**Safari:**
![Screenshot 2024-01-31 at 3 02 20 PM](https://github.com/iTwin/iTwinUI/assets/849817/e97072cd-e781-4491-aa48-4327b81a4b6e)

And I assume you are trying to avoid simply `padding: $surface-padding;` because if there is no scrollbar & then a scrollbar appears because of resizing or something, all of the body content will shift inwards, right?

Reply:
Yes, exactly.

Reply:
All of the content will *not* shift inwards as a result of resizing/content change because we are still using `scrollbar-gutter: stable` on the right edge. But yes I'm avoiding `padding: $surface-padding` because it does shift the content inwards too much when space is reserved for the scrollbar, especially when the content is not overflowing.

Reply:
Ahh, so trying to avoid this situation when there is no scrollbar present.
![Screenshot 2024-01-31 at 3 09 38 PM](https://github.com/iTwin/iTwinUI/assets/849817/fde0bbfe-21f5-4143-9792-d7de302e3201)

Reply:
Tried playing around with giving the scrollbar gutter a color (border would have been nicer), but when the content can't scroll it reserves the space, but hides the gutter color.  So this is not an option (also doesn't look good).

![Screenshot 2024-01-31 at 3 13 05 PM](https://github.com/iTwin/iTwinUI/assets/849817/99c52302-4a8c-499b-9aea-c07b0046638b)
![Screenshot 2024-01-31 at 3 13 15 PM](https://github.com/iTwin/iTwinUI/assets/849817/332bcf6e-3328-4f78-9e77-b3c029057e2e)

Reply:
Sounds good, leaving the code unchanged and updated PR description to mention this thread.